spec/index.html

```diff
@@ -1492,3 +1502,413 @@ <h1>SharedArray ( ..._values_ )</h1>
+          1. Set _newCV_.[[ConditionWaiterList]] to a new WaiterList Record.
+          1. Return ~unused~.
+        1. Return SharedStructCreate(_createInitializer_, « [[ConditionWaiterList]] »).
+      </emu-alg>
+    </emu-clause>
+  </emu-clause>
+
+  <emu-clause id="sec-properties-of-the-condition-constructor">
+    <h1>Properties of the Condition Constructor</h1>
+    <p>The Condition constructor:</p>
+    <ul>
+      <li>has a [[Prototype]] internal slot whose value is %Function.prototype%.</li>
+      <li>has the following properties:</li>
+    </ul>
+
+    <emu-note>
+      <p>Per-Realm prototypes, which is currently an open design question and not included in this draft, will give Conditions a per-Realm prototype with built-in methods instead of static methods.</p>
+    </emu-note>
+
+    <emu-clause id="sec-atomics.condition.wait">
+      <h1>Atomics.Condition.wait ( _cv_, _mutexUnlockToken_ )</h1>
+      <p>This function atomically unlocks _mutexUnlockToken_ and puts the surrounding agent in a wait queue and suspends it until the condition variable is notified.</p>
+      <p>It performs the following steps when called:</p>
+      <emu-alg>
+        1. Perform ? RequireInternalSlot(_cv_, [[ConditionWaiterList]]).
+        1. Perform ? RequireInternalSlot(_mutexUnlockToken_, [[LockedMutex]]).
+        1. Let _mutex_ be _mutexUnlockToken_.[[LockedMutex]].
+        1. If _mutex_ is ~empty~, throw a *TypeError* exception.
+        1. If AgentCanSuspend() is *false*, throw a *TypeError* exception.
+        1. Let _thisAgent_ be AgentSignifier().
+        1. Let _WL_ be _cv_.[[ConditionWaiterList]].
+        1. Perform EnterCriticalSection(_WL_).
+        1. Let _waiterRecord_ be a new Waiter Record { [[AgentSignifier]]: _thisAgent_, [[PromiseCapability]]: ~blocking~, [[TimeoutTime]]: +∞, [[Result]]: *"ok"* }.
+        1. Perform AddWaiter(_WL_, _waiterRecord_).
+        1. Perform UnlockMutex(_mutex_).
+        1. Perform SuspendThisAgent(_WL_, _waiterRecord_).
+        1. Perform LeaveCriticalSection(_WL_).
+        1. Perform LockMutex(_mutex_, +∞).
+        1. Assert: _waiterRecord_.[[Result]] is *"ok"*.
+        1. Return *undefined*.
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-atomics.condition.waitFor">
+      <h1>Atomics.Condition.waitFor ( _cv_, _mutexUnlockToken_, _timeout_ [ , _predicate_ ] )</h1>
+      <p>If _predicate_ is *undefined*, this function atomically unlocks _mutexUnlockToken_ and puts the surrounding agent in a wait queue and suspends it until the condition variable is notified or until the wait times out, returning *true* for the former and *false* for the latter.</p>
+      <p>If a _predicate_ is passed and calling it returns *false*, this function atomically unlocks _mutexUnlockToken_ and puts the surrounding agent in a wait queue and suspends it until _predicate_ returns *true*, or until the wait times out. The return value of the final call to _predicate_ is returned.</p>
```

Comment:
While it's detailed in the algorithm, it may be worth mentioning here that any time `predicate` is executing the condition will have reacquired a lock on the mutex (and presumably has reattached it to `mutexUnlockToken`).